### ISSUE TITLE
After swapping slots, the wrong slot's HTTP management webhook URLs may be returned for a while

### ISSUE THREAD

**Issue by gabesmsft** [Azure/azure-functions-durable-extension#557]
After doing a slot swap and then retrieving the HTTP management webhook URLs for a Orchestrator Durable Function, it can take a while (eg several minutes to several hours) for the URLs of the correct slot to be returned.

This seems to be correlated with the `WEBSITE_HOSTNAME` environment variable also taking a varying amount of time to update.

As we understand it, the `WEBSITE_HOSTNAME` environment variable can take a while to update to reflect the current slot by design, and there are several cases where the Durable Functions APIs depend on the `WEBSITE_HOSTNAME` environment to populate the management URLs.

Simple repro steps (keeping in mind the behavior isn't consistent):
1. Swap two slots that are using Durable Functions. Note: This was reproduced on V1 Function Apps.
2. Make a call to the Orchestrator on the production slot to get the management webhook URLs.
POST https://FakeFunctionApp.azurewebsites.net/api/OrchFunction?code=fakecode'

The response may return the staging slot's management webook URLs instead of the production slot's management webook URLs for a while before it updates to the correct urls.

Response body:
```json
{
    "id": "instanceid",
    "statusQueryGetUri": "https://FakeFunctionApp-staging.azurewebsites.net/admin/extensions/DurableTaskExtension/instances/instanceid?taskHub=taskhubid&connection=Storage&code=fakecode==",
    "sendEventPostUri": "https://FakeFunctionApp-staging.azurewebsites.net/admin/extensions/DurableTaskExtension/instances/instanceid/raiseEvent/{eventName}taskHub=taskhubid&connection=Storage&code=fakecode==",
    "terminatePostUri": "https://FakeFunctionApp-staging.azurewebsites.net/admin/extensions/DurableTaskExtension/instances/instanceid/terminate?reason={text}&taskHub=taskhubid&connection=Storage&code=fakecode==",
}
```

While the issue is happening, one has to manually change the URL to the production one in order to get the orchestration status of the production instance.

**Comment by cgillum** [comment]
In addition to checking whether `WEBSITE_HOSTNAME` is correctly set on the target VM, we need to also check whether this value is being inappropriately cached somewhere (functions host or the durable extension).

**Comment by ConnorMcMahon** [comment]
This was finally fixed in v2.4.2.
